### PR TITLE
Tests branch

### PR DESCRIPTION
**Pull request by chinweokwu**
This project was created on tests_branch. the branch contains the ruby specification test for ruby enumerables. RSpec test was written to check the behavior of the ruby enumerables that was created earlier on dev-branch.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+      it 'returns false if one or more values are falsy' do
+        expect([nil, true, 99].my_all?).to eq false
+      end
+    end
+  end
+
+  describe '.my_any?' do
+    context 'when receive a block' do
+      it 'returns true if one or more values follow the logic in block' do
```
Comment (Forison) [inline spec/enumerable_spec.rb]: * [x] Your test must cover the example below 
```[nil, true, 99].my_any?(Integer)                     #=> true```

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    context 'when no block is given' do
+      it 'returns an enumerator' do
+        result = array.my_select
+        expect(result).to be_an Enumerator
+      end
+    end
+  end
+
+  describe '.my_all?' do
+    context 'when receive a block' do
```
Comment (Forison) [inline spec/enumerable_spec.rb]: * [x] Your test for my_all should cover the cases below as well
``` %w[ant bear cat].my_all?(/t/)                        #=> false```
``` [1, 2i, 3.14].my_all?(Numeric)                       #=> true```
``` [ ].my_all?                                                     #=> true```

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      end
+
+      it 'returns false if one or more values are falsy' do
+        expect([].my_any?).to eq false
+      end
+    end
+  end
+
+  describe '.my_none?' do
+    context 'when receive a block' do
```
Comment (Forison) [inline spec/enumerable_spec.rb]: * [x] Let us make the test cover the examples below
  ```%w{ant bear cat}.my_none?(/d/)                        #=> true```
  ```[1, 3.14, 42].my_none?(Float)                         #=> false```
  ```[ ].my_none?                                           #=> true```
  ```[nil].my_none?                                        #=> true```

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+      it 'returns false if one or more values are truthy' do
+        expect([nil, false, true].my_none?).to eq false
+      end
+    end
+  end
+
+  describe '.my_count' do
+    context 'when receive an argument' do
+      it 'returns number of values that equals to the argument condition' do
```
Comment (Forison) [inline spec/enumerable_spec.rb]: * [x] Let us make our test cover the example below
```[ 1,2,3,4,5,6].count{ |x| x%2==0 } #=> 3```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes required☑️
# 🙋‍♂️ Team!!!
#### You have done a great job so far with your test, I am impressed. Your test covers many edge cases, however here are a few issues to address to get this project approved.

* [x] Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store)([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
* [x] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- Kindly add how to set up the project and run the test on a local machine
* [x] Define a variable for the test cases, students should [Use `let` instead](https://github.com/rubocop-hq/rspec-style-guide#let-blocks). ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/before_each.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
- ```array = [11, 2, 3, 56]```  should be define with let
### Check comments 👇below👇 for other cases that ought to be covered.
> Note: the test must cover all functionalities of each custom method
# That is it, kindly make the change and submit the project for review😎.
#### Happy coding💻
Add Forison

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### You have done great so far :clap: However, you have a few issues to address before your app can be approved.
- [ ] Your test cases are not passing 
test

### Make the changes and submit for another code review. Good luck.
- Happy coding!

By Raymond
slack @iamraymond

**Comment by Dandush03** [comment]
# Status: 🟢 Approved 🟢 

### Great Work Guys, Enjoy your next project

# Happy Coding 💻 

Daniel Laloush
